lesson26/src/js/login.js

```diff
@@ -0,0 +1,69 @@
+const userIdOfInput = document.querySelector(".js-form-userid");
+const passwordOfInput = document.querySelector(".js-form-password");
+const formElements = [userIdOfInput,passwordOfInput];
+const submitButton = document.getElementById("js-submit-button");
+
+const isNotEmptyOfInput = target => target.value.trim() !== "";
+const isNotEmptyOfAllInput = () => {
+    let result = false;
+    formElements.forEach(element => {
+        element.value.trim() !== "" ? result = true : result = false;
+    })
+    return result;
+}
+
+const checkFormValidityToEnableSubmitButton = () => submitButton.disabled = isNotEmptyOfAllInput() ? false : true;
+
+formElements.forEach(element => {
+    element.addEventListener("blur", (e) => {
+        checkFormValidityInBlur(e.target);
+    });
+});
+
+const checkFormValidityInBlur = target => {
+    submitButton.disabled = true;
+
+    if (!isNotEmptyOfInput(target)) {
+        target.nextElementSibling.textContent = "入力してください";
+        return;
+    }
+
+    target.nextElementSibling.textContent = "";
+    checkFormValidityToEnableSubmitButton();
+};
+
+const getInputValueAndTryLogin = () => {
+    const inputsValues = {
+        userId: document.querySelector(".js-form-userid").value,
+        password: document.querySelector(".js-form-password").value
+    }
+    tryToLogin(inputsValues);
+}
+
+const tryToLogin = async(inputsValues) => {
+    let result;
+    try {
+        result = await checkToRegistered(inputsValues);
+    } catch(rejectObj) {
+        result = rejectObj;
+    } finally {
+        result.token? window.location.href = "./loginuserpage.html" : window.location.href = "./notautherize.html";
+    }
```

Comment:
IMO
You can also write like this
```suggestion
        window.location.href = result.token ?  "./loginuserpage.html" :  "./notautherize.html";
    }
```

Reply:
Thanks for the review!
I see you have this way of writing!
It is smart. Thank you very much.
[bad1a27](https://github.com/haru-programming/JavaScript-lessons/pull/30/commits/bad1a27976c369715b3bd0d0dc64d866810a4a89)